lib/azure/armrest/armrest_service.rb

```diff
@@ -253,11 +254,22 @@ def rest_head(options)
 
         def raise_api_exception(err)
           begin
-            response = JSON.parse(err.http_body)
-            code     = response['error']['code']
-            message  = response['error']['message']
+            content_type_header = err.response.headers[:content_type]
+
+            response = case content_type_header.match(%r{(?<application_type>application\/\w+)})[:application_type]
```

Comment:
Oddly, I think this reads better:
```suggestion
            response = case content_type_header.match(%r{(application\/\w+)})[1]
```

Reply:
Also, if you're using %r, then there's no reason to escape the `/`

```suggestion
            response = case content_type_header.match(%r{(application/\w+)})[1]
```

OR

```suggestion
            response = case content_type_header.match(/(application\/\w+)/)[1]
```